File: public/dsa/arrays/cumulative_sum/naive/cumulative_sum.py
# imports-start
from typing import List

# imports-end


class Solution:
    def getCumulativeSum(self, arr: List[int]) -> List[int]:
        n = len(arr)
        result = [0 for _ in range(n)]
        for i in range(n):
            prefix_sum = 0
            for j in range(i + 1):
                prefix_sum += arr[j]
            result[i] = prefix_sum
        return result


# tests-start
sol = Solution()
arr = [1, 2, 3, 4]
sol.getCumulativeSum(arr)
# tests-end

'''output
[1, 3, 6, 10]
'''

File: public/dsa/arrays/cumulative_sum/naive/cumulative_sum.cpp

#include <bits/stdc++.h>
using namespace std;

vector<int> getCumulativeSum(vector<int> &arr) {
    int n = arr.size();
    vector<int> result(n);
    for (int i = 0; i < n; i++) {
        int prefixSum = 0;
        for (int j = 0; j <= i; j++) {
            prefixSum += arr[j];
        }
        result[i] = prefixSum;
    }
    return result;
}

// tests-start
int main() {
    vector<int> nums = {1, 2, 3, 4};
    auto result = getCumulativeSum(nums);
    for (int v : result) {
        cout << v << " ";
    }
    cout << endl;
    return 0;
}
// tests-end

/*output
1 3 6 10
*/